metadata-ingestion/src/datahub/ingestion/source/sql/druid.py

```diff
@@ -15,11 +17,25 @@
 from datahub.ingestion.source.sql.sql_config import BasicSQLAlchemyConfig
 
 
+def get_table_names(self, connection, schema=None, **kwargs):
+    query = "SELECT TABLE_NAME FROM INFORMATION_SCHEMA.TABLES"
+    if schema:
+        query = "{query} WHERE TABLE_SCHEMA = '{schema}'".format(
+            query=query, schema=schema
+        )
+
+    result = connection.execute(text(query))
+    return [row.TABLE_NAME for row in result] if result.rowcount > 0 else []
```

Comment:
What exactly is this `result.rowcount > 0` check doing? Does sqlalchemy normally error if the row count is 0?

Reply:
I threw an error (`This result object does not return rows. It has been closed automatically.`) if no rows were returned.

Reply:
This seems like a druid sqlalchemy bug to me... or something that should be configurable. Why would returning no rows result in an error?

Would at least prefer if we referenced the existing code (e.g. wrapped with a try catch) rather than overwriting it, if possible